src/EFCore.PG/Update/Internal/NpgsqlModificationCommandBatch.cs

```diff
@@ -51,16 +51,15 @@ public class NpgsqlModificationCommandBatch : ReaderModificationCommandBatch
             _maxBatchSize = maxBatchSize ?? DefaultBatchSize;
         }
 
-        protected override int GetParameterCount() => (int)_parameterCount;
+        protected override int GetParameterCount() => _parameterCount;
 
         protected override bool CanAddCommand(ModificationCommand modificationCommand)
         {
             if (ModificationCommands.Count >= _maxBatchSize)
                 return false;
 
-            var newParamCount = _parameterCount + modificationCommand.ColumnModifications.Count;
-
-            if (newParamCount > int.MaxValue)
+            var newParamCount = unchecked(_parameterCount + modificationCommand.ColumnModifications.Count);
```

Comment:
@YohDeadfall I'm not sure I get it. You seem to be purposefully overflowing here by doing int + int. Since the result type is int (newParamCount), it will be negative in that case. In the next line, since it's negative it will never be bigger than int.MaxValue (maybe you mean to put the uint case on the left side?).

Regardless, what are you trying to achieve with this change, and what exception were you referring to that the current code throws when compiled with checked arithmetic? _parameterCount's type is currently long precisely so that adding it with the new command's count wouldn't overflow int...

Reply:
On the next line was an error which fixed now. How it works? It really overflows an `int` and then compares it with the max value, but before it casts the new value to `uint` and does unsigned comparison.

Since counts and indices in .NET is 32-bit integers, I thought that `newParamCount` has `int` as type. Just forgot that `_parameterCount` is `long`. The previous version had no issues.

Therefore, it's a cosmetic fix.

Reply:
OK, this looks fixed now, but do you really think the new code is better than the old? IMHO it's clearer/more readable to just have a long, rather than deliberately overflow and then cast to uint, etc.

Reply:
BTW it's possible for _parameterCount to be int (obviating the cast in GetParameterCount), and to cast it to long only when adding it with modificationCommand.ColumnModifications.Count. That would clean the code up slightly.

Reply:
As I know .NET team uses `uint`s instead of `long`. Which way should we go?

Reply:
I prefer to keep things simple and avoid overflowing and "complicated" casting when we can avoid it... Remember that there is no perf concern here.

Reply:
Then it will look like following:

```csharp
var newParamCount = (long)_parameterCount + (long)modificationCommand.ColumnModifications.Count;
if (newParamCount > int.MaxValue)
    return false;

_parameterCount = (int)newParamCount;
```

As you can see this introduces a lot of conversions (one can be removed, but two others will survive). Is it okay?

Reply:
Unless I'm mistaken you can omit one of the long conversions (it happens implicitly). But yeah, that seems good to me.